Spatial-filtered Export failed under Oracle

The Export with the following query failed:
``` xml
<query xmlns="http://www.3dcitydb.org/importer-exporter/config">
  <typeNames>
    <typeName xmlns:bldg="http://www.opengis.net/citygml/building/2.0">bldg:Building</typeName>
  </typeNames>
  <filter>
    <intersects>
      <valueReference>bldg:boundedBy/bldg:GroundSurface/bldg:lod2MultiSurface</valueReference>
      <polygon>
        <exterior dimension="2">694173 5329252 694167 5329252 694149 5329250 694102 5329245 694173 5329252</exterior>
      </polygon>
    </intersects>
  </filter>
</query>
```

The error message is:

```
[07:55:18 ERROR] Failed to query the database.
[07:55:18 ERROR] Caused by: java.sql.SQLException: ORA-22901: VARRAY oder LOB-Attribute eines Objekttyps können nicht verglichen werden

[07:55:18 ERROR] Caused by: oracle.jdbc.OracleDatabaseException: ORA-22901: VARRAY oder LOB-Attribute eines Objekttyps können nicht verglichen werden

[07:55:18 WARN] Database export aborted.

```

--- a/impexp-core/src/main/java/org/citydb/core/query/builder/sql/SQLQueryBuilder.java
+++ b/impexp-core/src/main/java/org/citydb/core/query/builder/sql/SQLQueryBuilder.java
@@ -225,14 +225,20 @@ public class SQLQueryBuilder {
 	}
 
 	private void buildDistinctQuery(Query query, SQLQueryContext queryContext) {
-		if (!query.isSetSorting())
+		if (!query.isSetSorting() && !buildProperties.getAdditionalProjectionColumns().contains(MappingConstants.ENVELOPE)) {
 			queryContext.getSelect().setDistinct(true);
-
-		else {
+		} else {
 			// when sorting is enabled, then adding the sort column as projection token
 			// might lead to multiple rows per top-level feature even if distinct is used.
 			// so we have to rewrite the query using the row_number() function to guarantee
 			// that every top-level feature is only exported once.
+
+			// add order by id in case sorting has not been defined for the query
+			if (!query.isSetSorting()) {
+				queryContext.getSelect().addOrderBy(
+						new OrderByToken(queryContext.getFromTable().getColumn(MappingConstants.ID)));
+			}
+
 			Select inner = queryContext.getSelect();
 			List<ProjectionToken> projection = inner.getProjection();
 			List<OrderByToken> orderBy = inner.getOrderBy();
@@ -263,12 +269,20 @@ public class SQLQueryBuilder {
 					.addSelection(ComparisonFactory.equalTo(withTable.getColumn("rn_distinct"), new IntegerLiteral(1))));
 
 			// re-add projection columns to new select
-			for (ProjectionToken token : projection)
-				queryContext.getSelect().addProjection(token instanceof Column ? withTable.getColumn(((Column) token).getName()) : token);
+			for (ProjectionToken token : projection) {
+				queryContext.getSelect().addProjection(token instanceof Column ?
+						withTable.getColumn(((Column) token).getName()) :
+						token);
+			}
 
-			// re-add order by tokens to new select
-			for (int i = 0; i < orderBy.size(); i++)
-				queryContext.getSelect().addOrderBy(new OrderByToken(withTable.getColumn("order" + i), orderBy.get(i).getSortOrder()));
+			// re-add order by tokens to new select if sorting
+			// has been defined for the query
+			if (query.isSetSorting()) {
+				for (int i = 0; i < orderBy.size(); i++) {
+					queryContext.getSelect().addOrderBy(
+							new OrderByToken(withTable.getColumn("order" + i), orderBy.get(i).getSortOrder()));
+				}
+			}
 
 			queryContext.setFromTable(withTable);
 		}

